protocol.md

```diff
@@ -118,6 +118,7 @@ Following are SysEx commands used for core features defined in this version of t
 ```
 EXTENDED_ID                 0x00 // A value of 0x00 indicates the next 2 bytes define the extended ID
 RESERVED               0x01-0x0F // IDs 0x01 - 0x0F are reserved for user defined commands
+REPORT_FEATURES             0x65 // report which optional Firmata features are supported by the firmware
```

Comment:
I actually changed my mind while writing this...

I do like classing the features as either HARDWARE or FIRMWARE, because I don't see them as equal. I think our primary responsibility is to expose the hardware as hardware. Then, when limitations arise from the remote nature of the hardware, it becomes important to have FIRMWARE support; which I view as secondary. Theoretically, you can recreate any FIRMWARE you like if you are aware of the hardware, effectively leverage SYSEX messages, and are willing to modify the .ino (I know this is a crappy experience/solution).

So to summarize, I like HARDWARE versus FIRMWARE, instead of CORE vs EXTENDED. :smiley_cat: